plugins/woocommerce/includes/wc-order-step-logger-functions.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * WooCommerce Order Step Logging Functions
+ *
+ * Tracks the steps of the checkout process for place order debugging.
+ *
+ * @package WooCommerce\Functions
+ * @version 9.6.0
+ */
+
+declare(strict_types=1);
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Log an order-related message. This is not public API and should not be used by plugins or themes.
+ *
+ * @param string     $message Message to log.
+ * @param array|null $context Optional. Additional information for log handlers.
+ * @param bool       $final_step Optional. Whether this is the final step of the order logging, and should clear the log.
+ *
+ * @internal This function is intended for internal use only.
+ * @since 9.6.0
+ */
+function wc_log_order_step( string $message, ?array $context = null, bool $final_step = false ) {
```

Comment:
@senadir do we want to put this behind a toggle/flag for debugging? No right? I bet that most of the issues might be invisible to merchants, if shoppers are able to do the order again successfully or go buy somewhere else.

Reply:
>Do you mean only declaring the function
Preferably the best possible solution (returning early could be an option). 
Anyway doesn't matter, the question here was between this logging being an opt-in or not. For me it's not a question anymore, we're trying to catch unknown problems here, probably not even reported. We should ship it, wait, see. 👍🏼